File: java_integration/app/src/main/java/com/firewok/example/java_integration/feedresource/dynamiccontent/DynamicContentActivity.java
package com.firewok.example.java_integration.feedresource.dynamiccontent;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MenuItem;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import com.firework.common.feed.FeedResource;
import com.firewok.example.java_integration.BuildConfig;
import com.firewok.example.java_integration.R;
import com.firewok.example.java_integration.databinding.ActivityDynamicContentBinding;
import com.firework.videofeed.FwVideoFeedView;
import com.firework.viewoptions.BaseOption;
import com.firework.viewoptions.ViewOptions;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class DynamicContentActivity extends AppCompatActivity {
    private ActivityDynamicContentBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityDynamicContentBinding.inflate(LayoutInflater.from(this));
        setContentView(binding.getRoot());
        
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle(getString(R.string.dynamic_content_screen_title));
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayShowHomeEnabled(true);
        }

        setupDetails();
        initVideoFeedView();
    }

    @SuppressLint("SetTextI18n")
    private void setupDetails() {
        binding.details.source.setText("Dynamic");
        binding.details.channel.setText(BuildConfig.FW_CHANNEL_ID);
        binding.details.playlist.setText("N/A");
        binding.details.categories.setText("Test category");
    }

    private void initVideoFeedView() {
        FwVideoFeedView videoFeedView = binding.fwVideoFeedView;

        List<String> categories = Arrays.asList("Test category");
        Map<String, List<String>> parameters = new HashMap<>();
        parameters.put("category", categories);
        
        FeedResource.DynamicContent feedResource = new FeedResource.DynamicContent(
            BuildConfig.FW_CHANNEL_ID, 
            parameters
        );
        
        ViewOptions viewOptions = new ViewOptions.Builder()
            .baseOption(new BaseOption.Builder()
                .feedResource(feedResource)
                .build())
            .build();

        videoFeedView.init(viewOptions);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    public static Intent getIntent(Context context) {
        return new Intent(context, DynamicContentActivity.class);
    }
} 

File: feed_resources/app/src/main/java/tv/fw/example/dynamiccontent/DynamicContentActivity.kt
package tv.fw.example.dynamiccontent

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import androidx.appcompat.app.AppCompatActivity
import com.firework.common.feed.FeedResource
import com.firework.videofeed.baseOptions
import com.firework.videofeed.viewOptions
import tv.fw.example.feedResources.BuildConfig.FW_CHANNEL_ID
import tv.fw.example.feedResources.R
import tv.fw.example.feedResources.databinding.ActivityDynamicContentBinding

class DynamicContentActivity : AppCompatActivity() {

    private lateinit var binding: ActivityDynamicContentBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDynamicContentBinding.inflate(LayoutInflater.from(this))
        setContentView(binding.root)
        setTitle(R.string.dynamic_content_screen_title)

        setupDetails()

        initVideoFeedView()
    }

    @SuppressLint("SetTextI18n")
    private fun setupDetails() {
        binding.details.source.text = "Dynamic"
        binding.details.channel.text = FW_CHANNEL_ID
        binding.details.playlist.text = "N/A"
        binding.details.categories.text = "Test category"
    }

    private fun initVideoFeedView() {
        val videoFeedView = binding.fwVideoFeedView

        val categories = listOf("Test category")
        val parameters = mapOf("category" to categories)
        val feedResource = FeedResource.DynamicContent(channelId = FW_CHANNEL_ID, parameters = parameters)
        val viewOptions = viewOptions {
            baseOptions {
                feedResource(feedResource)
            }
        }
        videoFeedView.init(viewOptions)
    }

    companion object {
        fun intent(context: Context) = Intent(context, DynamicContentActivity::class.java)
    }
}
